ad9959/src/lib.rs

```diff
@@ -597,20 +599,16 @@ impl ProfileSerializer {
         self.index += value.len() + 1;
     }
 
+    #[inline]
     fn pad(&mut self) {
         // Pad the buffer to 32-bit (4 byte) alignment by adding dummy writes to CSR and LSRR.
-        match self.index & 3 {
-            3 => {
-                // For a level of 3, we have to pad with 5 bytes to align things.
-                self.add_write(Register::CSR, &[(self.mode as u8) << 1]);
-                self.add_write(Register::LSRR, &[0, 0]);
-            }
-            2 => self.add_write(Register::CSR, &[(self.mode as u8) << 1]),
-            1 => self.add_write(Register::LSRR, &[0, 0]),
-            0 => {}
-
-            _ => unreachable!(),
+        if self.index & 1 != 0 {
+            self.add_write(Register::LSRR, &[0, 0]);
```

Comment:
Mind adding some comments that adding an LSRR write pads us to even size-alignment, and then the CSR pads us to 4-byte (word) alignment?

Reply:
I added more comments explaining the strategy.